### PR TITLE
Performance Profiler: updates loading screen.

### PR DESCRIPTION
**Pull request by cpapazoglou**
Fixes https://github.com/Automattic/dotcom-forge/issues/8693

## Proposed Changes

Add the new loading screen from Figma i2.
Update to the next step every 5s and to another random tip every 10s.

## Testing Instructions

- Go to http://calypso.localhost:3000/speed-test-tool?url=https%3A%2F%2Fwww.quintoandar.com.br%2F&tab=mobile
- Check that loading screen appears
- Every 5 secs the step should change 
- Every 10s a random tip should be shown
- After 20-30 secs the dashboard content should be revealed

https://github.com/user-attachments/assets/2741612e-dc22-4bba-9def-8984d53937f4

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by gavande1 — COMMENTED** [review]
The loading screen looks amazing! Thanks for the great work. I have minor feedback that we should tackle in the next iteration. So when `hash` already exists in the URL, we should not show the loading screen. Instead, we should show the loading report progress bar since we are not actually running the test.

**Review by gavande1 — APPROVED** [review]

**Comment by WBerredo** [comment]
> The loading screen looks amazing! Thanks for the great work. I have minor feedback that we should tackle in the next iteration. So when hash already exists in the URL, we should not show the loading screen. Instead, we should show the loading report progress bar since we are not actually running the test.

I think we can tackle this in a follow up issue considering the test is being retrieved quickly enough now that is barely possible to read what is there.

File: client/performance-profiler/pages/loading-screen/index.tsx [hunk 1]
```diff
@@ -8,50 +9,238 @@ interface LoadingScreenProps {
...
+				translate( 'Fetching historic data' ),
+				translate( 'Identifying performance improvements' ),
+				translate( 'Finalizing your results' ),
 		  ];
 
-	const [ progress, setProgress ] = useState( 0 );
-	const [ tick, setTick ] = useState( 0 );
+	const tips = [
+		{
+			heading: translate( 'Performance Mattters' ),
```
Comment (emilyaudela) [inline client/performance-profiler/pages/loading-screen/index.tsx:153]: Assuming this is a typo (and not a play on Matt's name ;-) ). 

`Mattters` --> `Matters`

Comment (cpapazoglou) [reply client/performance-profiler/pages/loading-screen/index.tsx:153]: Thanks for the callout! Fixed in https://github.com/Automattic/wp-calypso/pull/93660

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16487105).

**Pull request merged**